File: apps/api-gateway/src/sentry-init.ts
/**
 * Optional Sentry for the API gateway. Call initSentry() after load-env; no-op when DSN is unset.
 */
import * as Sentry from "@sentry/node";
import { getGatewayEnv } from "@hyperagent/config";

export function initSentry(): void {
  const { sentry, nodeEnv } = getGatewayEnv();
  const dsn = sentry.dsn?.trim();
  if (!dsn) return;
  Sentry.init({
    dsn,
    environment: sentry.environment ?? nodeEnv ?? "development",
    tracesSampleRate: sentry.tracesSampleRate,
  });
}

export { Sentry };

File: services/orchestrator/sentry_init.py
"""Optional Sentry initialization for the orchestrator (set SENTRY_DSN)."""

from __future__ import annotations

import logging
import os

logger = logging.getLogger(__name__)


def init_sentry() -> None:
    dsn = os.environ.get("SENTRY_DSN", "").strip()
    if not dsn:
        return
    try:
        import sentry_sdk

        sentry_sdk.init(
            dsn=dsn,
            environment=os.environ.get(
                "SENTRY_ENVIRONMENT",
                os.environ.get("NODE_ENV", "development"),
            ),
            traces_sample_rate=float(os.environ.get("SENTRY_TRACES_SAMPLE_RATE", "0.1")),
        )
    except ImportError:
        logger.warning("sentry-sdk not installed; skipping Sentry (pip install sentry-sdk)")
